python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -841,8 +844,17 @@ def resolve_autoMaterializePolicy(
     def resolve_automationCondition(
         self, _graphene_info: ResolveInfo
     ) -> Optional[GrapheneAutoMaterializePolicy]:
-        if self._asset_node_snap.automation_condition:
-            return GrapheneAutomationCondition(self._asset_node_snap.automation_condition)
+        automation_condition = (
+            self._asset_node_snap.automation_condition_snapshot
+            or self._asset_node_snap.automation_condition
+        )
+        if automation_condition:
+            return GrapheneAutomationCondition(
+                # we only store one of automation_condition or automation_condition_snapshot
+                automation_condition
+                if isinstance(automation_condition, AutomationConditionSnapshot)
+                else automation_condition.get_snapshot()
+            )
```

Comment:
Don't see why we don't unconditionally use the snapshot in the graphql server

Reply:
As in why can't the asset_node_snap just have the snapshot?

Reply:
Responded to this above